libraries/AP_RSSI/AP_RSSI.cpp

```diff
@@ -128,6 +128,13 @@ float AP_RSSI::read_receiver_rssi()
         case RssiType::RSSI_RC_CHANNEL_VALUE :
             receiver_rssi = read_channel_rssi();
             break;
+        case RssiType::RSSI_RECEIVER : {
+            int16_t rssi = hal.rcin->get_rssi();
+            if (rssi != -1) {
+                receiver_rssi = rssi / 255.0;
+            }
```

Comment:
Here are some use cases that I've personally dealt with that the range value helps:

I experience some packet loss at very close range that I may personally elect not to label as meaningful loss of signal strength. So, for example, if I get 240/255 (94%) regularly, I may want to see that 94% clipped cleanly to 100%, so I change RECEIVER_PROTOCOL_HIGH.

Or, due to the inverse square law, it might turn out that the top 20% of packet loss is uninteresting to a particular user, since it only happens at short range, say 100-200 feet. I may decide I don't care and want to treat 204 (0.8 * 255) as RECEIVER_PROTOCOL_HIGH.

Finally, I've experienced radio systems where packets may be being received at a low level, but are effectively unflyable and in failsafe at that low rate. I may want to experience these as 0% RSSI.

If we are targeting this broadly at a number of vendors and protocols and systems, I don't think its safe to assume they will all do sane and expected things, and that any new ones that are clones or compatible ones will all do sane or expected things. Having the range values allows the user to not only tailor to their preferences, but also deal as best they can with problems in implementations.

I've had feedback from users about the original RSSI patch praising the ranges as a good implementation missing elsewhere, so I would hope we could keep this going with the new parameter.